src/lib/core/evio.c

```diff
@@ -205,24 +205,19 @@ evio_service_entry_accept_cb(ev_loop *loop, ev_io *watcher, int events)
  * listening on it. Unlink the file if it's the case.
  */
 static int
-evio_service_entry_reuse_addr(const char *uri)
+evio_service_entry_reuse_addr(const struct uri *u)
 {
-	struct uri u;
-	if (uri_create(&u, uri) || u.service == NULL) {
-		uri_destroy(&u);
-		diag_set(IllegalParams, "invalid uri for bind: %s", uri);
+	if (u->service == NULL) {
+		diag_set(IllegalParams, "invalid uri for bind");
```

Comment:
Nit: an assertion would do just fine, because the URI was checked in `box.cfg`. Same comment for `evio_service_entry_bind`.

Reply:
fixed